### PR TITLE
[T4A5][T09-B3]Clarence Chee

### PR DESCRIPTION
**Pull request by cheec**
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
+package seedu.addressbook.data.tag;
+
+import java.util.ArrayList;
+
+import seedu.addressbook.data.person.Person;
+
+public class Tagging {
+    
+    private final static ArrayList<Tagging> ALL_TAGGING_CHANGES = new ArrayList<Tagging>();
```
Comment (mauris) [inline src/seedu/addressbook/data/tag/Tagging.java:9]: this implementation would be case that was specified in the second diagram of the exercise question. Where else can be a better place to store all the tagging changes and why? 

Hint: look at the first diagram of the exercise question.

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    
+    private final static String ADDED_PREFIX   = "+";
```
Comment (mauris) [inline src/seedu/addressbook/data/tag/Tagging.java:11]: are there alternatives to constants? (:

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    
+    private final Tag    tag;
+    private final Person person;
+    private boolean      isLive; // whether tagging is live or deleted
+    
+    /**
+     * Creates a tagging object and records this creation in
+     * {@link #ALL_TAGGING_CHANGES}
+     */
+    public Tagging(Person person, Tag tag) {
```
Comment (mauris) [inline src/seedu/addressbook/data/tag/Tagging.java:27]: is there a way to straight away create a record of tagging that is meant for deletion? (i.e. can we set isLive straight from the constructor?)

File: src/seedu/addressbook/data/tag/Tagging.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    private final static String DELETED_PREFIX = "-";
+    
+    private final static String MESSAGE_TAGGING_CHANGES = "%1$s %2$s %3$s";
+    private final static String EXAMPLE_TAGGING_CHANGE  = "+ Jake Woo [friend]";
+    
+    private final static String LS = System.lineSeparator();
+    
+    private final Tag    tag;
+    private final Person person;
+    private boolean      isLive; // whether tagging is live or deleted
```
Comment (mauris) [inline src/seedu/addressbook/data/tag/Tagging.java:21]: if you have constants and other means of representing this tagging, you may want to consider what happens if there is just the state of "added" and "deleted"? (i.e. updated?)

**Comment by mauris** [comment]
Some comments added. Feel free to comment if you have any questions. Once done, you may the PR after reading comments.

**Comment by cheec** [comment]
another place to implement the tagging class is in the addressbook class itself. We can use enums instead of constants. you can record the state of the tagging in the constructor with an extra parameter. 

"if you have constants and other means of representing this tagging, you may want to consider what happens if there is just the state of "added" and "deleted"? (i.e. updated?)"  i dont understand this

**Comment by mauris** [comment]
@cheec, currently the state of the Tagging (added/deleted) is store as a boolean. Consider having to add "updated" state? Think about this (:

**Comment by cheec** [comment]
oh, perhaps a new TaggingState class or enums would suffice

**Pull request closed without merging**